File: post/src/main/java/com/code/post/model/Post.java
package com.code.post.model;


import com.fasterxml.jackson.annotation.JsonFormat;
import com.sun.istack.NotNull;

import javax.persistence.*;
import java.time.LocalDate;

@Entity
public class Post {

    @NotNull
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private String autor;

    @NotNull
    private String titulo;

    @NotNull
    @Lob
    private String texto;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "dd-MM-yyyy")
    private LocalDate data;

    public String getAutor() {
        return autor;
    }

    public void setAutor(String autor) {
        this.autor = autor;
    }

    public String getTitulo() {
        return titulo;
    }

    public void setTitulo(String titulo) {
        this.titulo = titulo;
    }

    public String getTexto() {
        return texto;
    }

    public void setTexto(String texto) {
        this.texto = texto;
    }

    public LocalDate getData() {
        return data;
    }

    public void setData(LocalDate data) {
        this.data = data;
    }
}

File: ng/post-web/src/app/models/post.ts
export class Post {
    id: string;
    autor: string;
    titulo: string;
    texto: string;
    data: string;

    constructor(autor: string, titulo: string, texto: string) {
        this.autor = autor;
        this.titulo = titulo;
        this.texto = texto;
    }

    setAutor(post: Post, novoAutor): void {
        post.autor = novoAutor;
    }

    setTitulo(post: Post, novoTitulo): void {
        post.titulo = novoTitulo;
    }

    setTexto(post: Post, novoTexto): void {
        post.texto = novoTexto;
    }
}
